.github/workflows/rust.yml

```diff
@@ -23,33 +20,35 @@ env:
   CARGO_TERM_COLOR: always
   RUST_BACKTRACE: 1
 
+permissions: {}
+
 jobs:
   test:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v5
+        with:
+          persist-credentials: false
       - name: Setup Rust
         run: rustup update --no-self-update
       - name: Install cargo-binstall
-        uses: cargo-bins/cargo-binstall@main
-      - name: Install cargo-llvm-cov and cargo-nextest
-        run: cargo binstall -y cargo-nextest cargo-llvm-cov
+        uses: cargo-bins/cargo-binstall@ae04fb5e853ae6cd3ad7de4a1d554a8b646d12aa # v1.15.11
+      - name: Install cargo-llvm-cov, cargo-nextest, and nur
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: cargo binstall -y cargo-nextest cargo-llvm-cov nur
```

Comment:
using nur is similar to using nushell. The version should be pinned in case an update breaks compatibility with the nurfile script.